tests/experimental/datasources/test_integration_sql_datasource.py

```diff
@@ -163,3 +165,110 @@ def test_run_checkpoint_and_data_doc(empty_data_context, include_rendered_conten
+    # We don't care about the exact number since that may change as data assistants evolve.
+    expectation_num = len(data_assistant_result.expectation_configurations)
+    assert expectation_num > 100
+    assert checkpoint_result.success, "Running expectation suite failed"
+    # Verify that the number of checkpoint validations is the number of expectations generated by the data assistant
+    assert len(checkpoint_result["run_results"][list(checkpoint_result["run_results"].keys())[0]]["validation_result"].results) == expectation_num
+
+
+pytest.mark.integration
+@pytest.mark.slow
+def test_run_multibatch_data_assistant_and_checkpoint(empty_data_context):
+    """Test using data assistants to create expectation suite and run checkpoint"""
+    data_assistant_result, checkpoint_result = configure_and_run_data_assistant_checkpoint(
+        context=empty_data_context,
+        sqlite_db_file_relative_path="../../test_sets/taxi_yellow_tripdata_samples/sqlite/yellow_tripdata_sample_2020_all_months_combined.db",
+        table_name="yellow_tripdata_sample_2020",
+        date_splitter_column="pickup_datetime",
+        batch_request_options={"year": 2020},
+    )
+    expectation_num = len(data_assistant_result.expectation_configurations)
+    assert expectation_num > 100
+    assert checkpoint_result.success, "Running expectation suite failed"
+    # Verify that the number of checkpoint validations is the number of expectations generated by the data assistant
+    assert len(checkpoint_result["run_results"][list(checkpoint_result["run_results"].keys())[0]]["validation_result"].results) == expectation_num
+
+
+def configure_and_run_data_assistant_checkpoint(
+        context: DataContext,
+        sqlite_db_file_relative_path: str,
+        table_name: str,
+        date_splitter_column: str,
+        batch_request_options: Dict[str, Any]
+):
+    db_file = file_relative_path(__file__, sqlite_db_file_relative_path)
+    # Add sqlalchemy datasource.
+    # The current method is called `add_postgres` it really should be `add_sql`.
+    # See https://superconductive.atlassian.net/browse/GREAT-1400
+    datasource = context.sources.add_postgres(
+        name="test_datasource",
+        connection_string=f"sqlite:///{db_file}",
+    )
+
+    # Add and configure a data asset
+    table = table_name
+    split_col = date_splitter_column
+    asset = (
+        datasource.add_table_asset(
+            name="my_asset",
+            table_name=table,
+        )
+        .add_year_and_month_splitter(column_name=split_col)
+        .add_sorters(["year", "month"])
+    )
+
+    # Setup an expectation suite using the onboarding data assistant
+    expectation_suite_name = "my_onboarding_assistant_suite"
+    context.create_expectation_suite(
+        expectation_suite_name=expectation_suite_name, overwrite_existing=True
+    )
+    batch_request = asset.get_batch_request(batch_request_options)
+    data_assistant_result = context.assistants.onboarding.run(
```

Comment:
@NathanFarmer How can I determine the number of batches that are part of the `data_assistant_result`? I know the results plotted by calling `data_assistant_result.pot_metrics()` has batches along the x axis so it seems like I can get this number.

Reply:
TLDR: Alex's offline comment is also what I would use. Something like: `len(DataAssistantResult._batch_id_to_batch_identifier_display_name_map)`.

The plots are generated one at a time at the Metric/Expectation-level by iterating through `DataAssistantResult.metrics_by_domain`. I don't think we implemented a clean method that does what you are wanting for the Data Assistant. Each domain should (in theory) have the same number of batch-metric values computed, but if a metric computation fails they could be different.

Reply:
@NathanFarmer @alexsherstinsky 
Probably not something to fix in this PR, but why doesn't `DataAssistantResult` implement `__len__`?

Reply:
@Kilo59 It isn't clear what the `__len__` would represent users IMO. Number of metrics computed? Number of Expectations suggested? Number of domains? Number of batches per domain?